internal/handlers/pg/pgdb/database_metadata.go

```diff
@@ -403,6 +421,47 @@ func (ms *metadataStorage) setIndex(ctx context.Context, index string, key Index
 	return
 }
 
+// checkExistingIndex checks if the given index already exists.
+//
+// It returns true and no error if existing and new index names and keys are equal.
+// It returns false and no error if existing and new index names and keys are different.
+// It returns false and an error if either name or key is equal (but not both).
+// Possible errors:
+//   - ErrIndexNameAlreadyExist - if index name already exists for a different key.
+//   - ErrIndexKeyAlreadyExist - if index key already exists for a different name.
+func checkExistingIndex(existing *metadataIndex, new *metadataIndex) (bool, error) {
+	var indexNameMatch bool
+
+	if existing.Name == new.Name {
+		indexNameMatch = true
+	}
+
+	if len(existing.Key) != len(new.Key) {
+		if indexNameMatch {
+			return false, ErrIndexNameAlreadyExist
+		} else {
+			return false, nil
+		}
```

Comment:
I think it can be simplified. Unless I'm reading it wrong. Applies to the rest of code as well
```suggestion
		}
		return false, nil
		
```